vertx-sql-client/src/main/java/io/vertx/sqlclient/Pool.java

```diff
@@ -164,6 +164,12 @@ default <T> void withTransaction(Function<SqlConnection, Future<@Nullable T>> fu
         .onComplete(ar -> conn.close()));
   }
 
+  /**
+   * Like {@link #withTransaction(Function, Handler)} but keeps the connection accessible via the context
+   * for the duration of the given {@code function}
+   */
+  <T> Future<@Nullable T> withPropagatedTransaction(Function<SqlConnection, Future<@Nullable T>> function);
```

Comment:
I think we should name this `withTransaction`.

Have you tried running the test case for this, with the existing withTransaction to see what is happening ?

Reply:
Agree, withPropagatedTransaction is not really pretty. However, we can't really name it withTransaction as the method signature collides with the existing one. Also, imho having distinguishable signatures seems like a good idea too to avoid confusion since the behaviour between the two differs.

I haven't done that, but I'll see if I get some time this weekend to look at it. Just mentioning it again that I have trouble running most of the tests. For now I've only been able to run the PG-tests as my dev environment doesn't seem to like the other ones.

Reply:
@vietj, I've tested a few different cases now by replacing withTransaction and withPropagatableTransaction with eachother.
Seems to be working fine;
- running withPropagatableTransaction inside the block of withTransaction results in another acquired connection and transaction. Inserting one entry using each of the two connection objects while also querying out the number of rows yields the expected result set size for each transaction inside their respective blocks.
- running withPropagatableTransaction, inserting something using its connection, querying out the result with the Pool.query (i.e. not using the aforementioned connection object) results in 0 rows, same goes withTransaction inside the same block.

The TransactionTestBase obviously needs to run with a larger pool size than 1 which it's currently set to. If you're fine with me increasing it then I'll gladly push a few more tests to verify the stuff mentioned above.

Reply:
I think this cas discussed a while ago and we could have a property for with transaction (e.g an enum TransactionMode) that would govern how nested `withTransaction` would behave, e.g `TransactionMode.PROPAGATABLE`.